### PR TITLE
Feature/change config managment

### PR DESCRIPTION
**Pull request by lschmid**
close #53, made updater as an ros package

### CODE REVIEW

**Comment by lschmid** [comment]
It does depend on the libvisensor 2.0.0

File: visensor_calibration_flasher/src/visensor_calibration_flasher.cpp [hunk 1]
```diff
@@ -112,63 +119,106 @@ int main(int argc, char **argv)
...
-    camera_calibration.R[0] = (double) T_C_I[0][0];
-    camera_calibration.R[3] = (double) T_C_I[0][1];
-    camera_calibration.R[6] = (double) T_C_I[0][2];
-    camera_calibration.R[1] = (double) T_C_I[1][0];
-    camera_calibration.R[4] = (double) T_C_I[1][1];
-    camera_calibration.R[7] = (double) T_C_I[1][2];
-    camera_calibration.R[2] = (double) T_C_I[2][0];
-    camera_calibration.R[5] = (double) T_C_I[2][1];
-    camera_calibration.R[8] = (double) T_C_I[2][2];
+    camera_calibration.R_.resize(9);
```
Comment (omaris) [inline visensor_calibration_flasher/src/visensor_calibration_flasher.cpp]: +1

File: visensor_calibration_flasher/src/visensor_calibration_flasher.cpp [hunk 2]
```diff
@@ -112,63 +119,106 @@ int main(int argc, char **argv)
...
+      projection_model->principal_point_v_ = (double) intrinsics[3];
+    }
+    else if (camera_model == std::string("omnidirectional")){
+      visensor::ViCameraProjectionModelOmnidirectional::Ptr projection_model = camera_calibration.getProjectionModel<visensor::ViCameraProjectionModelOmnidirectional>();
+      projection_model->setType();
+      projection_model->focal_length_u_ = (double) intrinsics[0];
+      projection_model->focal_length_v_ = (double) intrinsics[1];
+      projection_model->principal_point_u_ = (double) intrinsics[2];
+      projection_model->principal_point_v_ = (double) intrinsics[3];
+      // todo(lschmid) how to check if intrinsics has 5 attributes?
```
Comment (omaris) [inline visensor_calibration_flasher/src/visensor_calibration_flasher.cpp]: I think accessing an XmlRpc::XmlRpcValue out of bounds will throw exception. Please double check

Comment (omaris) [inline visensor_calibration_flasher/src/visensor_calibration_flasher.cpp]: Check out http://xmlrpcpp.sourceforge.net/doc/classXmlRpc_1_1XmlRpcValue.html#a30 . Should be what youre looking for, I think

File: visensor_calibration_flasher/src/visensor_calibration_flasher.cpp [hunk 3]
```diff
@@ -112,63 +119,106 @@ int main(int argc, char **argv)
...
-    std::string distortion_model;
 
-    camera_model.assign(cam_params["camera_model"]);
-    distortion_model.assign(cam_params["distortion_model"]);
+    camera_calibration.slot_id_ = 0;
+
+
+    // delete every factroy calibration of the corresponding cam
+    try {
+      privat_drv->cleanCameraCalibrations(camera_id, 0, -1,
```
Comment (omaris) [inline visensor_calibration_flasher/src/visensor_calibration_flasher.cpp]: As mentioned in libvisensor pull request -> Can we have a function such as cleanCameraCalibrations(camera_id) which wraps this function call?

File: visensor_calibration_flasher/CMakeLists.txt [hunk 4]
```diff
@@ -3,12 +3,13 @@ project(visensor_calibration_flasher)
...
-  libvisensor
 )
 
-include_directories(include ${catkin_INCLUDE_DIRS})
+find_package(libvisensor 2.0.0 REQUIRED)
+  
+include_directories(include ${catkin_INCLUDE_DIRS} ${libvisensor_INCLUDE_DIRS})
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native -Wall -std=c++0x -D__STRICT_ANSI__")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native -Wall -Wno-reorder -std=c++0x -D__STRICT_ANSI__")
```
Comment (lschmid) [inline visensor_calibration_flasher/CMakeLists.txt]: remove warnings do not ignore them

File: visensor_calibration_flasher/src/visensor_calibration_flasher.cpp [hunk 5]
```diff
@@ -32,25 +32,28 @@
 #include <ros/ros.h>
 #include <ros/package.h>
 
-#include <visensor/visensor.hpp>
+#include <visensor_impl.hpp>
 
 void printSensorConfig(const visensor::ViCameraCalibration& config){
-
+  visensor::ViCameraLensModelRadial::Ptr lens_model = config.getLensModel<visensor::ViCameraLensModelRadial>();
```
Comment (lschmid) [inline visensor_calibration_flasher/src/visensor_calibration_flasher.cpp]: made printing independent of the lens_model

File: visensor_update/src/SensorUpdater.cpp [hunk 6]
```diff
@@ -30,137 +30,124 @@
...
 
-//function returns a vector of pairs with (package_name, version)
-//for all packages that start with the given packagename prefix
-bool SensorUpdater::getVersionInstalled(SensorUpdater::VersionList &outPackageList, const std::string &prefix, bool dontParseVersion)
+/**
+ * function returns a vector of pairs with (package_name, version)
+ * for all packages that start with the given packagename prefix
+ */
+bool SensorUpdater::getVersionInstalled(VersionList* outPackageList, const std::string& prefix,
+bool dontParseVersion)
```
Comment (diemf) [inline visensor_update/src/SensorUpdater.cpp]: indent

File: visensor_update/src/SensorUpdater.cpp [hunk 7]
```diff
@@ -489,39 +427,237 @@ bool SensorUpdater::installPackagesFromPath(SensorUpdater::VersionList &packageL
...
+
+bool SensorUpdater::convertCalibration()
+{
+  visensor::ViSensorConfiguration::Ptr config_server = boost::make_shared<
+      visensor::ViSensorConfiguration>(pFile_transfer_);
+  std::cout << "Convert calibration to new format ... ";
+
+  std::string tmp_calibration_filename("/tmp/calibration.xml");
+  if (!loadXmlCameraCalibrationFile(tmp_calibration_filename)) {
+    std::cout << "no calibration file was found, assume that the sensor is not yet calibrate"
```
Comment (diemf) [inline visensor_update/src/SensorUpdater.cpp]: ... not yet calibrated

File: visensor_update/src/SensorUpdater.cpp [hunk 8]
```diff
@@ -489,39 +427,237 @@ bool SensorUpdater::installPackagesFromPath(SensorUpdater::VersionList &packageL
...
+              << std::endl;
+    return true;
+  }
+
+  // try to load existing configuration already saved in the new format
+  try {
+    config_server->loadConfig();
+  } catch (visensor::exceptions const &ex) {
+    std::cout << "ignore" << std::endl
+              << "no new configurations were found, assume that the sensor has no" << std::endl;
```
Comment (diemf) [inline visensor_update/src/SensorUpdater.cpp]: ... the sensor has none

File: visensor_update/src/SensorUpdater.cpp [hunk 9]
```diff
@@ -489,39 +427,237 @@ bool SensorUpdater::installPackagesFromPath(SensorUpdater::VersionList &packageL
...
+    std::cout << "Exception was: " << ex.what() << std::endl;
+  }
+
+  std::vector<visensor::ViCameraCalibration> calibration_list = parseXmlCameraCalibration(
+      tmp_calibration_filename);
+  if (calibration_list.size() == 0) {
+    std::cout << "failed\n";
+    std::cout << "no calibration were found" << std::endl;
```
Comment (diemf) [inline visensor_update/src/SensorUpdater.cpp]: no calibrations were found

File: visensor_update/src/main.cpp [hunk 10]
```diff
@@ -43,6 +45,7 @@ void printArgs(void)
     std::cout << "  Available commands are:" << std::endl;
     std::cout << "     update               updates the sensor to the newest software on the online repo" << std::endl;
     std::cout << "     update-16488         updates the sensor with ADIS 16488 to the newest software on the online repo" << std::endl;
+    std::cout << "     convert-calibration  converts the calibration format from libvisensor version 1.2.X to 2.0.X"  << std::endl;
     std::cout << "     clean                removes all software on the sensor" << std::endl;
     std::cout << "     version              shows installed packages " << std::endl;
     std::cout << "     reboot               reboot sensor " << std::endl;
```
Comment (diemf) [inline visensor_update/src/main.cpp]: is the "download to local path" and "update from local path" not yet implemented?

**Comment by diemf** [comment]
scolgtm

File: visensor_update/src/main.cpp (hunk 10 above)
Comment (lschmid) [inline visensor_update/src/main.cpp]: this functionality is implemented in another pull request

**Pull request merged**